src/io/index.js

```diff
@@ -0,0 +1,308 @@
+            });
+        });
+    }
+
+    // Uploads array of files into filesystem
+    async createFilesFromArrayBuffer(files) {
+        const promises = files.map(async file => {
+            file.buffer = Object.values(file.buffer);
+            const result = await this.createFileFromArrayBuffer(file);
+            return result;
+        });
+
+        return await Promise.all(promises);
+    }
+
+    // Deletes everything in Path
+    async deletePathRecursively(path) {
+        return new Promise((resolve, reject) => {
+            fs.stat(path, function(err, stats) {
+                if (err) return reject(format404(path));
+
+                // If this is a dir, begin recursive deletion
+                if (stats.isDirectory()) {
+                    sh.rm(path, { recursive: true }, function(err) {
+                        if (err) return reject(err);
+                        resolve(`Successfully removed ${path}.`);
+                    });
+                } else {
+                    fs.unlink(path, function(err) {
+                        if (err) return reject(err);
+                        resolve(`Successfully removed ${path}.`);
+                    });
+                }
+            });
+        });
+    }
+
+    // Deletes a given Directory /A
+    async deleteDirectory(path) {
+        return new Promise((resolve, reject) => {
+            fs.rmdir(path, function(err) {
+                if (err) return reject(err);
+                resolve();
+            });
+        });
+    }
+
+    // Deletes a given File /A.txt
+    async deleteFile(path) {
+        return new Promise((resolve, reject) => {
+            fs.unlink(path, function(err) {
+                if (err) return reject(err);
+                resolve();
+            });
+        });
+    }
+
+    // Format filesystem
+    async clearFileSystem() {
+        return new Promise((resolve, reject) => {
+            formatFS(function(err) {
+                if (err) return reject(err);
+                resolve();
+            });
+        });
+    }
+
+    // Creates a given Path /A
+    async createPath(path) {
+        return new Promise((resolve, reject) => {
+            sh.mkdirp(path, function(err) {
+                if (err) return reject(err);
+                resolve();
+            });
+        });
+    }
+
+    // Retrieves a file
+    async getFile(path) {
+        // TODO: need to add promises to Filer
+        return new Promise((resolve, reject) => {
+            fs.stat(path, (err, stats) => {
+                if (err) {
+                    return reject(err);
+                } else {
+                    // If this is a dir, show a dir listing
+                    if (stats.isDirectory()) {
+                        // Todo: Better error handling needed.
+                        return reject('Path does not link to a File.');
+                    } else {
+                        fs.readFile(path, (err, contents) => {
```

Comment:
Probably best to just try and `fs.readFile` and it will return an `EISDIR` error if the path is a directory, which you can handle in the UI calling this.